tests/dragonfly/conftest.py

```diff
@@ -112,8 +112,8 @@ def parse_args(args: List[str]) -> Dict[str, Union[str, None]]:
     return args_dict
 
 
-@pytest.fixture(scope="function", params=[{}])
-def df_factory(request, tmp_dir, test_env) -> DflyInstanceFactory:
+@pytest_asyncio.fixture(scope="function", params=[{}])
+async def df_factory(request, tmp_dir, test_env) -> DflyInstanceFactory:
```

Comment:
This is the bread and butter of the PR. Fixtures can be async! This allows us to:

1. Save all client we create on df instance
2. Clean them at the end of the fixture (since we can now `await close()`)